packages/alfa-rules/src/sia-r75/rule.ts

```diff
@@ -28,16 +32,22 @@ export default Rule.Atomic.of<Page, Element>({
   uri: "https://alfa.siteimprove.com/rules/sia-r75",
   tags: [Scope.Component, Stability.Stable],
   evaluate({ device, document }) {
+    const visibleTextCache = Cache.empty<Element<string>, Sequence<Text>>();
+
     return {
       applicability() {
         return getElementDescendants(document, Node.fullTree).filter(
           and(
             hasNamespace(Namespace.HTML),
             not(hasName("sup", "sub")),
-            Node.hasTextContent(),
-            isVisible(device),
-            // If the font-size ultimately computes to size 0, the element is not
-            // visible.
+            (node) =>
+              visibleTextCache
+                .get(node, () =>
+                  node
+                    .descendants(Node.fullTree)
+                    .filter(and(isText, isVisible(device)))
+                )
+                .some(() => true),
```

Comment:
```suggestion
            not((node) =>
              visibleTextCache
                .get(node, () =>
                  node
                    .descendants(Node.fullTree)
                    .filter(and(isText, isVisible(device)))
                )
                .isEmpty()),
```
Makes the intention clearer.